sycl/include/CL/sycl/detail/sycl_mem_obj_t.hpp

```diff
@@ -9,15 +9,15 @@
 #pragma once
 
 #include <CL/sycl/detail/common.hpp>
-#include <CL/sycl/detail/memory_manager.hpp>
-#include <CL/sycl/detail/scheduler/scheduler.hpp>
+#include <CL/sycl/detail/sycl_mem_obj_allocator.hpp>
 #include <CL/sycl/detail/sycl_mem_obj_i.hpp>
 #include <CL/sycl/detail/type_traits.hpp>
 #include <CL/sycl/event.hpp>
 #include <CL/sycl/property_list.hpp>
 #include <CL/sycl/stl.hpp>
 
 #include <type_traits>
+#include <cstring>
```

Comment:
I believe cstring should be placed before type_traits

Reply:
I believe that suggested placement is correct. See [LLVM Coding Standards](https://llvm.org/docs/CodingStandards.html#include-style): there are 4 main sections of includes, each of them is sorted alphabetically, but they didn't overlap between each other

Reply:
@AlexeySachkov I have moved `cstring` above `type_traits`. For some reason GitHub doesn't show this comment as outdated.